### PR TITLE
RFC-0003 `PortableDid` & `web5` CLI

### PR DESCRIPTION
**Pull request by KendallWeihe**
Add RFC-0003 which includes the addition of the `PortableDid`; please read the RFC for full context.

Implemented a basic `web5` CLI (which makes use of the `PortableDid` -- see the README for sample usage.

---

Also fixed a small bugs in `crates/web5/src/dids/methods/did_dht/bep44.rs`, `crates/web5/src/crypto/dsa/ed25519.rs`, and `crates/web5/src/dids/data_model/document.rs`.

Also fixed RFC-0002 which was accidentally left behind.

### CODE REVIEW

File: docs/API_DESIGN.md [hunk 1]
```diff
@@ -682,9 +685,85 @@ CLASS BearerDid
...
 > [!WARNING]
 >
-> We'll need to add support for the developer to select a VM other than defaulting to the first in the `verification_method` array; add `METHOD get_signer_by_kid(key_id: string): Signer`.
\ No newline at end of file
+> We'll need to add support for the developer to select a VM other than defaulting to the first in the `verification_method` array; add `METHOD get_signer_by_kid(key_id: string): Signer`.
+
+### Example: Instantiate from a [`PortableDid`](#portabledid)
+
+```pseudocode!
+portable_did = JSON.parse(env.get("PORTABLE_DID"))
```
Comment (KendallWeihe) [inline docs/API_DESIGN.md:699]: we have a `CONSTRUCTOR(json: string)` for good reason (to give more useful error results), so we should use it here!

**Comment by KendallWeihe** [comment]
Need to merge #255 and #256 prior to merging this, but won't do so until review is completed because it will bloat the review of the changes here

File: Cargo.toml [hunk 2]
```diff
@@ -1,8 +1,9 @@
 [workspace]
 members = [ 
+    "crates/cli",
```
Comment (nitro-neal) [inline Cargo.toml:3]: i wonder if we should name it web5_cli to keep the theme going, and then maybe one day there will be a tbdex_cli

File: crates/cli/src/dids/create.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+impl Commands {
+    pub fn command(&self) {
+        match self {
+            Commands::Jwk {
+                no_indent,
+                json_escape,
+            } => {
+                let private_jwk = Ed25519Generator::generate();
+                let mut public_jwk = private_jwk.clone();
+                public_jwk.d = None;
```
Comment (nitro-neal) [inline crates/cli/src/dids/create.rs:54]: yea these are strange, more data points that we need a public/private Jwk 'lite' type

File: crates/web5/src/crypto/dsa/ed25519.rs [hunk 4]
```diff
@@ -37,7 +37,7 @@ pub(crate) fn public_jwk_from_bytes(public_key: &[u8]) -> Result<Jwk> {
     }
 
     Ok(Jwk {
-        alg: "EdDSA".to_string(),
+        alg: "Ed25519".to_string(),
```
Comment (nitro-neal) [inline crates/web5/src/crypto/dsa/ed25519.rs:40]: you sure this is right?
image

image

I'm a bit confused on this issue at this point, but pretty much all other sources have an ed25519 jwk the alg set to EdDSA

File: docs/API_DESIGN.md [hunk 5]
```diff
@@ -682,9 +685,85 @@ CLASS BearerDid
...
+    "capabilityInvocation": [
+      "did:dht:4nca8jd5q5qwrowbx1efrihkac6danj6fpkhrrnrhdifiq19xfry#0"
+    ],
+    "capabilityDelegation": [
+      "did:dht:4nca8jd5q5qwrowbx1efrihkac6danj6fpkhrrnrhdifiq19xfry#0"
+    ]
+  },
+  "privateKeys": [
+    {
+      "alg": "Ed25519",
```
Comment (nitro-neal) [inline docs/API_DESIGN.md:761]: if you update the above, reupdate these

File: web5 [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+cargo build -p cli > /dev/null 2>&1
+cp ./target/debug/cli ./target/debug/web5
```
Comment (nitro-neal) [inline web5:4]: should the path still be debug?

**Review by nitro-neal — APPROVED** [review]
few small things but overall ✅

File: crates/web5/src/crypto/dsa/ed25519.rs (hunk 4 above)
Comment (KendallWeihe) [reply crates/web5/src/crypto/dsa/ed25519.rs:40]: yup we're moving to fully-specific alg https://github.com/TBD54566975/web5-rs/issues/128

File: Cargo.toml (hunk 2 above)
Comment (KendallWeihe) [reply Cargo.toml]: agreed, done

**Pull request merged**